lib/hammer_cli_katello/content_view_version.rb

```
@@ -259,6 +261,214 @@ class ExportCommand < HammerCLIKatello::SingleResourceCommand
+
+      command_name "import"
+
+      success_message _("Content view imported.")
+      failure_message _("Could not import the content view")
+
+      option(
+        '--export-tar',
+        'EXPORT_TAR',
+        _("Location of export tar on disk.")
+      )
+
+      validate_options do
+        option(:option_export_tar).required
+      end
+
+      build_options
+
+      def execute
+        fail unless File.exist?(options['option_export_tar'])
+
+        self.export_tar_file = File.basename(options['option_export_tar'])
+        self.export_tar_prefix = @export_tar_file.gsub('.tar', '')
+        self.export_tar_dir = File.dirname(options['option_export_tar'])
+        untar_export
+
+        cv = content_view(export_json['name'])
+        sync_repositories
+        repo_units = export_json['repositories'].collect { |repo| repo['rpm_filenames'] }.uniq
+
+        publish(cv['id'], export_json['major'], export_json['minor'], repo_units)
+      end
+
+      def untar_export
+        Dir.chdir(@export_tar_dir) do
+          `tar -xf #{@export_tar_file}`
+        end
+
+        Dir.chdir("#{@export_tar_dir}/#{@export_tar_prefix}") do
+          if File.exist?("#{@export_tar_file.gsub('.tar', '-repos.tar')}")
+            `tar -xf #{@export_tar_file.gsub('.tar', '-repos.tar')}`
+          else
+            fail "Export repos tar file is missing"
+          end
+        end
+      end
+
+      def export_json
+        json_file = "#{@export_tar_dir}/#{@export_tar_prefix}/#{@export_tar_file.gsub('tar', 'json')}"
+        @export_json ||= JSON.parse(File.read(json_file))
+      end
+
+      def sync_repositories
+        @export_json['repositories'].each do |repo|
+          synchronize(repo['id'], "file://#{@export_tar_dir}/#{@export_tar_prefix}/#{repo['relative_path']}")
+        end
+      end
+
+      def content_view(name)
+        HammerCLIForeman.foreman_resource(:content_views).call(:index, 'name' => name)['results'].first
```

Silently fails if you have multiple orgs with the same content view names :(